support/vm/invocation_context.go

```diff
@@ -183,22 +214,31 @@ func (ic *invocationContext) StateTransaction(obj cbor.Er, f func()) {
 }
 
 func (ic *invocationContext) VerifySignature(signature crypto.Signature, signer address.Address, plaintext []byte) error {
+	charge, err := ic.gasPrices.OnVerifySignature(signature.Type, len(plaintext))
+	if err != nil {
+		return err
+	}
+	ic.chargeGas(charge)
 	return ic.Syscalls().VerifySignature(signature, signer, plaintext)
 }
 
 func (ic *invocationContext) HashBlake2b(data []byte) [32]byte {
+	ic.chargeGas(ic.gasPrices.OnHashing(len(data)))
 	return ic.Syscalls().HashBlake2b(data)
 }
 
 func (ic *invocationContext) ComputeUnsealedSectorCID(reg abi.RegisteredSealProof, pieces []abi.PieceInfo) (cid.Cid, error) {
+	ic.chargeGas(ic.gasPrices.OnComputeUnsealedSectorCid(reg, pieces))
 	return ic.Syscalls().ComputeUnsealedSectorCID(reg, pieces)
 }
 
 func (ic *invocationContext) VerifySeal(vi proof.SealVerifyInfo) error {
+	ic.chargeGas(ic.gasPrices.OnVerifySeal(vi))
 	return ic.Syscalls().VerifySeal(vi)
 }
 
 func (ic *invocationContext) BatchVerifySeals(vis map[address.Address][]proof.SealVerifyInfo) (map[address.Address][]bool, error) {
+	// no explicit gas charged
```

Comment:
See https://github.com/filecoin-project/specs-actors/pull/1381#discussion_r629873262, I hope we can remove the special case.